app/lib/proxy_config_affecting_changes.rb

```diff
@@ -122,13 +122,18 @@ def destroy
         super
       end
 
-      protected
+      def write_attribute(attr_name, value)
+        track_proxy_affecting_changes if proxy_config_affecting_attributes.include?(attr_name.to_s)
+        super
+      end
```

Comment:
Well, not sure, because then I guess on `super` the overridden `_write_attribute` will be called instead of the `write_attribute`.

Reply:
What is this class for? If I look for `write_attribute` on the code base I can't find any usage.

Reply:
@jlledom Yes, we don't call this method explicitly, it's an internal Rails method that it called when updating attributes.

It is tricky, and I don't like it, because the internal methods may change anytime (and this is what happened with this upgrade).

Basically, this module "intercepts" attribute updates on the models that include it, and "tracks" this changes, so we can show to the user that their proxy/service configuration has changed and they need to update their staging APIcast configuration in order to apply these changes.

I am not a fan of this implementation, but I don't want to change it at this point.

Reply:
So, here for example, is how `_write_attribute` is called, when we do `service.update(name: 'UPP')`:
![image](https://github.com/3scale/porta/assets/1270328/1de8b4dd-4f33-45fa-b175-667c2fcb72bc)

Reply:
Yeah, in [6.0](https://github.com/rails/rails/blob/v6.0.6.1/activerecord/lib/active_record/attribute_methods/write.rb) we had `#write_attribute` calling `#_write_attribute`. But [now and until 7.1](https://github.com/rails/rails/blob/v7.1.3.2/activerecord/lib/active_record/attribute_methods/write.rb) both call `@attributes.write_from_user` directly which is IMO a needless change. Just one item in the call stack. I guess it saves a couple of nanoseconds now and then...

Although somebody claimed 15% perf boost https://github.com/rails/rails/commit/27a1ca2bfeda4298bbf44da17d07fac4147a4b1c